gon/linear.py

```diff
@@ -88,9 +88,7 @@ def __eq__(self, other: 'Segment') -> bool:
         False
         """
         return (self._start == other._start and self._end == other._end
-                or self._start == other._end and self._end == other._start
-                if isinstance(other, Segment)
-                else NotImplemented)
+                or self._start == other._end and self._end == other._start)
```

Comment:
I believe the previous implementations of `__eq__` methods were more error-proof.